server/src/routes/api/v1/sitesRouter.js

```diff
@@ -1,26 +1,54 @@
 import express from "express";
+import uploadImage from "../../../services/uploadImage.js";
+import cleanUserInput from "../../../services/cleanUserInput.js";
 import { Site } from "../../../models/index.js";
+import { ValidationError } from "objection";
+import SiteSerializer from "../../../serializers/SiteSerializer.js"
 
 const sitesRouter = new express.Router();
 
 sitesRouter.get("/", async (req, res) => {
     try {
         const sites = await Site.query();
-        return res.status(200).json({ siteList: sites });
+        const serializedSites = SiteSerializer.getSummary(sites)
+
+        return res.status(200).json({ siteList: serializedSites });
+    } catch (error) {
+        return res.status(500).json({ errors: error });
+    }
+});
+
+sitesRouter.post("/", uploadImage.single("image"), async (req, res) => {
+    try {
+        const { body, user } = req;
+        const formInput = cleanUserInput(body);
+
+        const data = {
+            ...formInput,
+            image: req.file?.location,
+            creatorId: user.id
+        };
+
+        const newSite = await Site.query().insertAndFetch(data);
+        return res.status(201).json({ site: newSite });
     } catch (error) {
+        console.log(error)
```

Comment:
console log - one I asked for while debugging, but we should look for and remove before submitting a PR!